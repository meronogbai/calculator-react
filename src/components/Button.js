import PropTypes from 'prop-types';

export default function Button({ name, clickHandler }) {
  const handleClick = () => clickHandler(name);
  const getClass = name => {
    switch (name) {
      case '0':
        return 'Button zero';
      case '+':
      case '-':
      case 'X':
      case 'รท':
      case '=':
        return 'Button operator';
      default:
        return 'Button';
    }
  };
  return (
    <button type="button" onClick={handleClick} className={getClass(name)}>
      {name}
    </button>
  );
}

Button.propTypes = {
  name: PropTypes.string.isRequired,
  clickHandler: PropTypes.func.isRequired,
};
