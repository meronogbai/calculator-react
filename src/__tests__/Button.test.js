import { create } from 'react-test-renderer';
import Button from '../components/Button';

test('renders correctly', () => {
  const handleClick = () => { ''; };
  const tree = create(<Button clickHandler={handleClick} name="" />).toJSON();
  expect(tree).toMatchSnapshot();
});
