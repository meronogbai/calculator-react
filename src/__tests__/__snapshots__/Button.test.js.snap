// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="Button"
  onClick={[Function]}
  type="button"
>
  
</button>
`;
